include/inviwo/core/datastructures/bitset.h

```diff
@@ -0,0 +1,399 @@
+using void_t = void;
+
+template <class T, class = void>
+struct is_iterator : std::false_type {};
+
+template <class T>
+struct is_iterator<T, void_t<typename std::iterator_traits<T>::iterator_category>>
+    : std::true_type {};
+
+}  // namespace detail
+
+/**
+ * \brief represents a bitset based on roaring bitmaps provided by the CRoaring library
+ */
+class IVW_CORE_API BitSet {
+public:
+    class BitSetIterator {
+    public:
+        using iterator_category = std::bidirectional_iterator_tag;
+        using value_type = uint32_t;
+        using difference_type = uint32_t;
+        using pointer = const uint32_t*;
+        using reference = const uint32_t&;
+
+        BitSetIterator() = default;
+        ~BitSetIterator();
+
+        BitSetIterator& operator++();
+        BitSetIterator operator++(int);
+        BitSetIterator& operator--();
+        BitSetIterator operator--(int);
+
+        value_type operator*() const;
+
+        bool operator==(const BitSetIterator& rhs) const;
+        bool operator!=(const BitSetIterator& rhs) const;
+
+    private:
+        using RoaringIt = roaring::RoaringSetBitForwardIterator;
+
+        friend BitSet;
+        BitSetIterator(const BitSet& b, bool exhausted = false);
+        BitSetIterator(const RoaringIt& it);
+        BitSetIterator(const BitSetIterator& rhs);
+        BitSetIterator& operator=(const BitSetIterator& rhs);
+
+        std::unique_ptr<RoaringIt> it_;
+    };
+
+    BitSet();
+
+    BitSet(const std::vector<uint32_t>& values);
+
+    template <size_t N>
+    BitSet(const std::array<uint32_t, N>& values) : BitSet() {
+        addMany(values.size(), values.data());
+    }
+
+    template <typename InputIt,
+              class = typename std::enable_if_t<detail::is_iterator<InputIt>::value>>
+    explicit BitSet(InputIt begin, InputIt end) : BitSet() {
+        add(begin, end);
+    }
+
+    template <typename... uint32_ts>
+    explicit BitSet(uint32_t val, uint32_ts&&... values) : BitSet() {
+        add(val, values...);
+    }
+
+    BitSet(const BitSet& rhs);
+    BitSet(BitSet&& rhs) noexcept;
+    BitSet& operator=(const BitSet& rhs);
+    BitSet& operator=(BitSet&& rhs) noexcept;
+    ~BitSet() = default;
+
+    BitSetIterator begin() const;
+    BitSetIterator end() const;
+
+    /**
+     * return the number of elements the bitset holds
+     */
+    uint32_t cardinality() const;
+
+    bool empty() const;
+
+    void clear();
+
+    /**
+     * check whether this bitset is a subset of \p b
+     */
+    bool isSubsetOf(const BitSet& b) const;
+    /**
+     * check whether this bitset is a strict subset of \p b
+     */
+    bool isStrictSubsetOf(const BitSet& b) const;
+
+    /**
+     * Add value \p v to the bitset
+     *
+     * @param v  value
+     */
+    void add(uint32_t v);
+
+    void add(const std::vector<uint32_t>& values);
+
+    template <size_t N>
+    void add(const std::array<uint32_t, N>& values) {
+        addMany(values.size(), values.data());
+    }
```

Comment:
just use span, no need for any other containers

Reply:
does span then use implicit converters for containers?

Reply:
yes